VoodooI2C/VoodooI2C/VoodooI2CDevice/VoodooI2CDeviceNub.cpp

```diff
@@ -148,16 +148,31 @@ IOReturn VoodooI2CDeviceNub::getAlternativeInterrupt(VoodooI2CACPICRSParser* crs
 
     data->release();
 
-    if (!crs_parser->found_gpio_interrupts) {
-        IOLog("%s::%s Failed to find valid interrupts from _DSM or XDSM method\n", getName(), acpi_device->getName());
+    IOLog("%s::%s Found valid resources from _CRS method\n", getName(), acpi_device->getName());
+
+    return kIOReturnSuccess;
+}
+
+IOReturn VoodooI2CDeviceNub::parseResourcesDSM(VoodooI2CACPICRSParser* crs_parser) {
+    OSObject *result = nullptr;
+    OSData *data = nullptr;
+    if (getDeviceResourcesDSM(TP7G_FUNC_INDEX, &result) != kIOReturnSuccess ||
+        !(data = OSDynamicCast(OSData, result))) {
+        IOLog("%s::%s Could not retrieve resources from _DSM or XDSM method\n", getName(), acpi_device->getName());
+        OSSafeReleaseNULL(result);
         return kIOReturnNotFound;
     }
 
-    IOLog("%s::%s Found valid interrupts from _DSM or XDSM method\n", getName(), acpi_device->getName());
+    uint8_t const* crs = reinterpret_cast<uint8_t const*>(data->getBytesNoCopy());
+    crs_parser->parseACPICRS(crs, 0, data->getLength());
+
+    data->release();
```

Comment:
I much prefer to use `OSSafeReleaseNULL` macro here instead of calling release directly.

Reply:
Done